fuse/test/file_lock_test.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2016 the Go-FUSE Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build linux
+
+package test
+
+import (
+	"bytes"
+	"os"
+	"os/exec"
+	"syscall"
+	"testing"
+	"path/filepath"
+
+	"github.com/hanwen/go-fuse/fuse"
+	"github.com/hanwen/go-fuse/fuse/nodefs"
+	"github.com/hanwen/go-fuse/internal/testutil"
+)
+
+func TestFlockExclusive(t *testing.T) {
+	cmd, err := exec.LookPath("flock")
+	if err != nil {
+		t.Skip("flock command not found.")
+	}
+	tc := NewTestCase(t)
+	defer tc.Cleanup()
+
+	contents := []byte{1, 2, 3}
+	tc.WriteFile(tc.origFile, []byte(contents), 0700)
+
+	f, err := os.OpenFile(tc.mountFile, os.O_WRONLY, 0)
+	if err != nil {
+		t.Fatalf("OpenFile(%q): %v", tc.mountFile, err)
+	}
+	defer f.Close()
+
+	if err = syscall.Flock(int(f.Fd()), syscall.LOCK_EX); err != nil {
+		t.Errorf("Flock returned: %v", err)
+		return
+	}
+
+	if out, err := runExternalFlock(cmd, tc.mountFile); !bytes.Contains(out, []byte("failed to get lock")) {
+		t.Errorf("runExternalFlock(%q): %s (%v)", tc.mountFile, out, err)
```

Comment:
use %q iso. %s. For []byte %s will use [1 2 3] as formatting.

Reply:
ping

Reply:
This test was slightly changed. Since on Ubuntu Trusty flock does not have a verbose option we can't assert an output. So we relaxed an assertion that command fails which in turn removed this `Errorf` line. 

https://github.com/hanwen/go-fuse/pull/220/files#diff-49306d0ec901b97786aec218f18ee349R21